### PR TITLE
Fixed alpha preservation when converting 4-channel Mat to UIImage

### PR DESCRIPTION
**Pull request by themightyoarfish**
I hope this is not too naively done, but MatToUIImage needs to preserve the alpha channel.

### CODE REVIEW

**Comment by vpisarev** [comment]
:+1:

**Comment by themightyoarfish** [comment]
Strange. I am unable to build the Framework on OSX at all, so I cannot diagnose the problem (the log output being very unhelpful). It compiles in XCode obviously and does not change the interface so I can't really see where the build errors are coming from.

File: modules/imgcodecs/src/ios_conversions.mm [hunk 1]
```diff
@@ -73,8 +77,7 @@
                                         8 * image.elemSize(),
                                         image.step.p[0],
                                         colorSpace,
-                                        kCGImageAlphaNone|
-                                        kCGBitmapByteOrderDefault,
+                                        bitmapInfo,
```
Comment (berak) [inline modules/imgcodecs/src/ios_conversions.mm]: bitMapInfo, not bitmapInfo

File: modules/imgcodecs/src/ios_conversions.mm [hunk 2]
```diff
@@ -66,6 +66,10 @@
     CGDataProviderRef provider =
             CGDataProviderCreateWithCFData((__bridge CFDataRef)data);
 
+    // Preserve alpha transparency, if exists
+    bool alpha = cvMat.channels() == 4;
```
Comment (berak) [inline modules/imgcodecs/src/ios_conversions.mm]: there is no cvMat in your function. did you mean 'image' ?

Comment (themightyoarfish) [inline modules/imgcodecs/src/ios_conversions.mm]: Indeed, this happens when copy-pasting and not being able to test…

**Comment by themightyoarfish** [comment]
I've fixed the errors, will the build be reattempted automatically?

**Comment by alalek** [comment]
Yes (just update your branch).
Builds status is here: http://pullrequest.opencv.org/#/summary/opencv

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**